packages/ketcher-react/src/script/editor/Editor.ts

```diff
@@ -613,7 +622,30 @@ function updateLastCursorPosition(editor: Editor, event) {
   }
 }
 
-function domEventSetup(editor: Editor, clientArea) {
+function toolIsUsed(
```

Comment:
One small proposal – may be it is worth to rename a function to `useToolIfNeeded`?
And then
```
const isToolUsed = useToolIfNeeded(...)
if (isToolUsed) {
  return true
}
```

Not 100% sure it is better, but at the moment the function `toolIsUsed` doesn't only check the condition (as the name implies), but also has side effect – it applies the tool